smarts/core/sensors.py

```diff
@@ -238,6 +248,22 @@ def observe(sim, agent_id, sensor_state, vehicle):
             else None
         )
 
+        near_via_points = []
+        hit_via_point = []
+        nearest_remaining_via_point = None
+        if vehicle.subscribed_to_via_point_sensor:
```

Comment:
How about adding the `via_point_data` to observation if there is any, i.e., without subscribing to it.

Reply:
I will do this. :ok:

Reply:
Having some trouble doing this, for now it can just be registered by not in the `AgentInterface`.